File: orthodb/writeOrthoDbProjectXML.pl
#!/usr/bin/perl

use strict;
use warnings;

use Getopt::Std;
use XML::LibXML;

use lib '../modules/perl';
require ImProjectXml;

use feature ':5.12';

my $usage = "Usage:writeOrthoDbProjectXML.pl [-h] [-i <project-xml-path>] <data-path> <taxon-id-path>

synopsis: read the taxon id file and creates an InterMine project XML <source> entry
If a <project-xml-path> is given then this entry is inserted directly into the given project XML
otherwise it is printed on stdout

options:
\t-h\tthis usage
";

my (%opts, $insert);

getopts('hi:', \%opts);
defined $opts{"h"} and die $usage;
defined $opts{"i"} and $insert++;
my $insertPath = $opts{i};

@ARGV > 1 or die $usage;

my ($dataPath, $taxonIdPath, $projectXmlPath) = @ARGV;

open TAXONS, $taxonIdPath or die "Could not open $taxonIdPath: $!";
my $taxonIds = <TAXONS>;
chomp($taxonIds);
close TAXONS;

my $sourceXml 
  = ImProjectXml::generateSource(
    "orthodb", "orthodb", 1,
    [
      [ 'src.data.dir', 'location', $dataPath ],
      [ 'orthodb.organisms', $taxonIds ]
    ]);

if ($insert) {
  ImProjectXml::insertSourceIntoProjectXml($insertPath, $sourceXml);
} else {
  say $sourceXml;
}

File: orthodb/writeOrthoDbProjectXml.py
#!/usr/bin/env python

import argparse
import os
import sys
import shutil

sys.path.insert(1, os.path.dirname(os.path.abspath(__file__)) + '/../modules/python')
import intermine.project as imp

class Logger(object):
    def __init__(self, logPath):
        self.terminal = sys.stdout
        self.log = open(logPath, "a")

    def write(self, message):
        self.terminal.write(message)
        self.log.write(message)

class MyParser(argparse.ArgumentParser):
    def error(self, message):
        sys.stderr.write('error: %s\n' % message)
        self.print_help()
        sys.exit(2)

############
### MAIN ###
############
parser = MyParser('Add OrthoDB project entry to InterMine SynBioMine project XML.')
parser.add_argument('datasetPath', help='path to the dataset location.')
parser.add_argument('orthoDbDataPath', help='path to the OrthoDB data')
parser.add_argument('-v', '--verbose', action="store_true", help="verbose output")
args = parser.parse_args()

datasetPath = args.datasetPath
orthoDbDataPath = args.orthoDbDataPath

logPath = "%s/logs/writeOrthoDbProjectXml.log" % datasetPath
sys.stdout = Logger(logPath)
projectXmlPath = "%s/intermine/project.xml" % datasetPath
taxonsPath = "%s/taxons/taxons.txt" % datasetPath

with open(taxonsPath) as f:
    taxons = f.read()

project = imp.Project("%s/intermine/project.xml" % datasetPath)
source = imp.Source(
    'orthodb', 'orthodb',
    [
        { 'name':'src.data.dir', 'location':orthoDbDataPath },
        { 'name':'orthodb.organisms', 'value':taxons }
    ],
    dump=True)

project.addSource(source)
print project.toString()

shutil.copy(projectXmlPath, "%s.bak" % projectXmlPath)
project.write(projectXmlPath)